Account/FxAPushMessageHandler.swift

```diff
@@ -94,33 +97,33 @@ extension FxAPushMessageHandler {
                         case .tabReceived(_, let tabData):
                             let title = tabData.entries.last?.title ?? ""
                             let url = tabData.entries.last?.url ?? ""
-                            deferred.fill(Maybe(success: PushMessage.commandReceived(tab: ["title": title, "url": url])))
+                            completion(.success(PushMessage.commandReceived(tab: ["title": title, "url": url])))
                         }
                     case .deviceConnected(let deviceName):
-                        deferred.fill(Maybe(success: PushMessage.deviceConnected(deviceName)))
+                        completion(.success(PushMessage.deviceConnected(deviceName)))
                     case let .deviceDisconnected(deviceId, isLocalDevice):
                         if isLocalDevice {
                             // We can't disconnect the device from the account until we have access to the application, so we'll handle this properly in the AppDelegate (as this code in an extension),
                             // by calling the FxALoginHelper.applicationDidDisonnect(application).
                             self.profile.prefs.setBool(true, forKey: PendingAccountDisconnectedKey)
-                            deferred.fill(Maybe(success: PushMessage.thisDeviceDisconnected))
+                            completion(.success(PushMessage.thisDeviceDisconnected))
                         }
 
                         guard let profile = self.profile as? BrowserProfile else {
                             // We can't look up a name in testing, so this is the same as not knowing about it.
-                            deferred.fill(Maybe(success: PushMessage.deviceDisconnected(nil)))
+                            completion(.success(PushMessage.deviceDisconnected(nil)))
                             break
                         }
 
                         profile.getClient(fxaDeviceId: deviceId).uponQueue(.main) { result in
```

Comment:
I briefly tried to go further and tackle the deferred being returned here and the one on the `reconfig` in line 72 but that was a bit thorny as the main app depended on those APIs - That all should be cleaned up when Profile and RustFirefoxAccounts move out of deferred